src/test/java/com/github/checkstyle/regression/internal/GitUtils.java

```diff
@@ -75,8 +80,23 @@ public static File addAnEmptyFileAndCommit(Repository repository, String fileNam
     public static void addAllAndCommit(Repository repository, String message)
             throws GitAPIException {
         try (Git git = new Git(repository)) {
+            final Status status = git.status().call();
```

Comment:
In git CLI, `git add .` add all kinds of change, including deleted. But in JGit, the `add .` could not add the deleted files into staging area. To obtain the same behavior as git CLI, we have to use `RmCommand` to handle deleted files.